Userland/Libraries/LibHTTP/HttpRequest.cpp

```diff
@@ -182,9 +182,21 @@ Optional<HttpRequest> HttpRequest::from_raw_request(ReadonlyBytes raw_request)
     else
         return {};
 
-    request.m_resource = URL::percent_decode(resource);
     request.m_headers = move(headers);
 
+    String requested_url = URL::percent_decode(resource);
```

Comment:
:thinking: this doesn't sound like something that should be done in LibHTTP, but rather the applications using it

Reply:
Make sense, as for your comment: removed it from LibHTTP and moved it to WebServer. Not gonna lie I didn't feel comfortable with messing around the path handling part inside the WebServer request handling. I did some testing to verify that I didn't introduce any new bugs(hacky stuff like `GET /%2e%2e/%2e%2e/home/anon/secret.txt`). As far as I tested, it looks like the `LexicalPath::join` does its magic correctly bc I made sure to decode the resource beforehand and not after building the final path. However, I would appreciate/suggest reviewing it carefully since it's my 1st PR and i'm not familiar with the whole serenity ecosystem. Any feedback would be appreciated, lmk your thoughts.

Reply:
Looks alright, we can always fix it later if it causes some difficult to notice issue